web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -76,7 +80,21 @@ export function UnifiedResources(props: {
   const { clustersService } = useAppContext();
   const { userPreferencesAttempt, updateUserPreferences, userPreferences } =
     useUserPreferences(props.clusterUri);
-  const { documentsService, rootClusterUri } = useWorkspaceContext();
+  const { documentsService, rootClusterUri, accessRequestsService } =
+    useWorkspaceContext();
+  const addedResources = useStoreSelector(
+    'workspacesService',
+    useCallback(
+      state => {
+        const pending =
+          state.workspaces[rootClusterUri]?.accessRequests.pending;
+        if (pending?.kind === 'resource') {
+          return pending.resources;
+        }
+      },
+      [rootClusterUri]
+    )
```

Comment:
This does make me a bit uneasy, but I guess this selector is fine overall? The identity is preserved, it's just that sometimes we return undefined. 🤔

Reply:
Yeah, I think we can't get away from returning undefined, since the kind can be `role`.